x-pack/plugins/triggers_actions_ui/public/application/components/update_api_key_modal_confirmation.tsx

```diff
@@ -49,22 +81,24 @@ export const UpdateApiKeyModalConfirmation = ({
         setUpdateModalVisibility(false);
         setIsLoadingState(true);
         try {
-          await Promise.all(idsToUpdate.map((id) => apiUpdateApiKeyCall({ id, http })));
-          toasts.addSuccess(
-            i18n.translate('xpack.triggersActionsUI.updateApiKeyConfirmModal.successMessage', {
-              defaultMessage:
-                'API {idsToUpdate, plural, one {key} other {keys}} {idsToUpdate, plural, one {has} other {have}} been updated',
-              values: { idsToUpdate: idsToUpdate.length },
-            })
-          );
+          if (bulkApiUpdateApiKeyCall) {
+            const response = await bulkApiUpdateApiKeyCall({
```

Comment:
for simplicity can we just use the `bulkApiUpdateApiKeyCall` why do we need the other one? is it because they can remove id by unselected rules when they go though the pagination of the rule table.

Reply:
Done

Reply:
make sense to me, let's create a tech debt issue